main.tf

```
@@ -0,0 +1,133 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
+# Create a VPC
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+# Local variables
+locals {
+  public_cidr = ["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr = ["10.0.2.0/24", "10.0.3.0/24"]
+}
+# Create a Public Subnet
+resource "aws_subnet" "public" {
+    count = 2
```

This will only work if we have 2 subnets. However, the task says "at least 2" which means we could have 3,4 or even 5 public/private subnets.
Instead of hardcoding 2, use the example below:
```suggestion
  count = length(local. public_cidr)
```

You can read more about the. "length" function in the following guide:
https://www.terraform.io/language/functions/length

---

fixed